source/main/dqm4hep-mysql-dump-parameters.cc

```diff
@@ -0,0 +1,164 @@
+ * Parses the different directories and populate the
+ * qtest file with found objects.
+ */
+
+// -- dqm4hep headers
+#include <dqm4hep/StatusCodes.h>
+#include <dqm4hep/Internal.h>
+#include <dqm4hep/DBInterface.h>
+#include <dqm4hep/tinyxml.h>
+
+// -- tclap headers
+#include "tclap/CmdLine.h"
+#include "tclap/Arg.h"
+
+using namespace std;
+using namespace dqm4hep::core;
+
+struct ParameterEntry
+{
+  std::string      m_parameter;
+  std::string      m_value;
+  std::string      m_lastUpdate;
+};
+
+typedef std::vector<ParameterEntry> ParameterEntryList; 
+
+int main(int argc, char* argv[])
+{
+  std::string cmdLineFooter = "Please report bug to <dqm4hep@gmail.com>";
+  std::unique_ptr<TCLAP::CmdLine> pCommandLine(new TCLAP::CmdLine(cmdLineFooter, ' ', DQMCore_VERSION_STR));
+
+  TCLAP::ValueArg<std::string> xmlOutputFileArg(
+      "o"
+      , "output-file"
+      , "The xml output file"
+      , false
+      , ""
+      , "string");
+  pCommandLine->add(xmlOutputFileArg);
+  
+  TCLAP::ValueArg<std::string> dbHostArg(
+      "k"
+      , "host"
+      , "The database host"
+      , false
+      , "localhost"
+      , "string");
+  pCommandLine->add(dbHostArg);
+  
+  TCLAP::ValueArg<std::string> dbDbArg(
+      "d"
+      , "database"
+      , "The database name"
+      , false
+      , "DQM4HEP"
+      , "string");
+  pCommandLine->add(dbDbArg);
+  
+  TCLAP::ValueArg<std::string> dbTableArg(
+      "t"
+      , "table"
+      , "The database table to dump"
+      , true
+      , ""
+      , "string");
+  pCommandLine->add(dbTableArg);
+
+  // parse command line
+  pCommandLine->parse(argc, argv);
+  
+  
+  DBInterface interface;
+
+  try
+  {
+    THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, interface.connect(dbHostArg.getValue(), "DQM4HEP", "", dbDbArg.getValue()));
+    THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, interface.dumpParameterTable(dbTableArg.getValue()));
+  }
+  catch(StatusCodeException &exception)
+  {
+    dqm_error( "MySQL error, couldn't dump parameters from table: {0}", exception.toString() );
+    return 1;
+  }
+  
+  if(xmlOutputFileArg.isSet())
+  {
+    StringMap parameterMap;
+    
+    try
+    {
+      THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, interface.getTableParameters(dbTableArg.getValue(), parameterMap));
+    }
+    catch(StatusCodeException &exception)
+    {
+      dqm_error( "MySQL error, couldn't dump parameters from table: {0}", exception.toString() );
+      return 1;
+    }
+        
+    std::string outputFile(xmlOutputFileArg.getValue());      
+    TiXmlDocument document;
+    
+    TiXmlDeclaration *decl = new TiXmlDeclaration( "1.0", "", "" );
+    document.LinkEndChild( decl );
+    
+    TiXmlElement *root = new TiXmlElement("dqm4hep");
+    document.LinkEndChild( root );
+    
+    // database info
+    TiXmlElement *databases = new TiXmlElement("databases");
+    root->LinkEndChild( databases );
+    
+    TiXmlElement *db = new TiXmlElement("db");
+    db->SetAttribute("id", "mydb");
```

Comment:
We could pass the dbName and user as parameters on the cmdline too

Reply:
The db name is already customizable by command line. For the user, it is always "DQM4HEP" as a read only user always defined.